src/reader.c

```diff
@@ -215,25 +208,19 @@ static int Reader_init(hiredis_ReaderObject *self, PyObject *args, PyObject *kwd
         if (!_Reader_set_exception(&self->replyErrorClass, replyErrorClass))
             return -1;
 
-    if (encodingObj) {
-        PyObject *encbytes;
-        char *encstr;
-        int enclen;
-
-        if (PyUnicode_Check(encodingObj))
-            encbytes = PyUnicode_AsASCIIString(encodingObj);
-        else
-            encbytes = PyObject_Bytes(encodingObj);
+    self->encoding = encoding;
+    if(errors) {
+        if (strcmp(errors, "strict") != 0 &&
+            strcmp(errors, "replace") != 0 &&
+            strcmp(errors, "ignore") != 0 &&
+            strcmp(errors, "backslashreplace") != 0) {
 
```

Comment:
`errors='surrogateescape'` should also be allowed (PEP 383, supported since python 3.1)